File: Top_interview_150/26-Remove-Duplicates-from-Sorted-Array.cpp
class Solution {
public:
    int removeDuplicates(vector<int>& nums) {
        if(nums.size()==1)return 1;

        int val=nums[0];
        int count=1;
        int i=1,j=1;
        while(j<nums.size()){
            if(nums[j]>val){
                val=nums[j];
                swap(nums[i],nums[j]);
                i++;
                count++;
            }
            j++;
        }
        return count;
    }
};

File: Random/Arrays/26-Remove-Duplicates-from-Sorted-Array.ts
function removeDuplicates(nums: number[]): number {
    const len=nums.length;
    let j=1;
    let k=1;
    for (let i=1;i<len;i++){
        if(nums[i]!==nums[i-1]){
            nums[j]=nums[i];
            j++;
            k++;
        }
    }
    return k;
};

